File: Parallel and Distributed Programming/Lab0-Java/src/main/java/Adunare/CalculatorSerial.java
package Adunare;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class CalculatorSerial {

    private List<Integer> numarMare1;
    private List<Integer> numarMare2;

    private List<Integer> vectorRezultat = new ArrayList<>();
    private List<Integer> vectorCarry = new ArrayList<>();

    public CalculatorSerial(List<Integer> numarMare1, List<Integer> numarMare2) {

        this.vectorCarry.add(0);
        this.numarMare1 = numarMare1;
        this.numarMare2 = numarMare2;

        Collections.reverse(numarMare1);
        Collections.reverse(numarMare2);

        this.checkVectors();
    }

    private void checkVectors(){
        while (numarMare1.size() > numarMare2.size())
            numarMare2.add(0);

        while (numarMare2.size() > numarMare1.size())
            numarMare1.add(0);
    }

    public List<Integer> aduna(){
        for (int i = 0; i < this.numarMare1.size(); i++) {
            this.vectorRezultat.add((this.numarMare1.get(i) + this.numarMare2.get(i))%10);
            if(this.numarMare1.get(i) + this.numarMare2.get(i) > 9)
                this.vectorCarry.add(1);
            else this.vectorCarry.add(0);
        }
        List<Integer> rezultat = new ArrayList<>();
        for(int i = 0; i < this.vectorRezultat.size(); i++){
            rezultat.add((this.vectorRezultat.get(i)+this.vectorCarry.get(i))%10);
            if(this.vectorRezultat.get(i)+this.vectorCarry.get(i) > 9)
                this.vectorCarry.set(i+1,this.vectorCarry.get(i+1)+1);
        }

        if(this.vectorCarry.get(this.vectorCarry.size()-1) == 1)
            rezultat.add(1);

        Collections.reverse(rezultat);
        return rezultat;
    }
}

File: Parallel and Distributed Programming/Lab0-C++/Lab0-C++/CalculatorSerial.cpp
#pragma once
#include "CalculatorSerial.h"

void CalculatorSerial::checkVectors() {
	while (numarMare1.size() > numarMare2.size())
		numarMare2.push_back(0);
	while (numarMare1.size() < numarMare2.size())
		numarMare1.push_back(0);
}

std::vector<int> CalculatorSerial::aduna() {
	for (int i = 0; i < numarMare1.size(); i++) {
		vectorRezultat.push_back((numarMare1.at(i) + numarMare2.at(i)) % 10);
		if (numarMare1.at(i) + numarMare2.at(i) > 9)
			vectorCarry.push_back(1);
		else vectorCarry.push_back(0);
	}
	std::vector<int> rezultat;
	for (int i = 0; i < vectorRezultat.size(); i++) {
		rezultat.push_back((vectorRezultat.at(i) + vectorCarry.at(i)) % 10);
		if (vectorRezultat.at(i) + vectorCarry.at(i) > 9)
			vectorCarry.at(i + 1) = vectorCarry.at(i + 1) + 1;
	}

	if (vectorCarry.at(vectorCarry.size() - 1) == 1)
		rezultat.push_back(1);

	std::reverse(std::begin(rezultat), std::end(rezultat));
	return rezultat;
}